Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -61,63 +69,75 @@ void DataReaderTask::run(ProcessingContext& pc)
+
+  for (const auto& ref : InputRecordWalker(pc.inputs(), dummy)) {
+    const auto dh = o2::framework::DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+    if (dh->payloadSize == 0) { //}|| dh->payloadSize==16) {
+      LOGP(WARNING, "Found blank input input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : ",
+           dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      sendData(pc, true); //send the empty tf data.
+      return;
+    }
+    LOG(info) << " matched DEADBEEF";
+  }
+  //TODO combine the previous and subsequent loops.
   /* loop over inputs routes */
   for (auto iit = pc.inputs().begin(), iend = pc.inputs().end(); iit != iend; ++iit) {
     if (!iit.isValid()) {
       continue;
     }
     /* loop over input parts */
+    int inputpartscount = 0;
     for (auto const& ref : iit) {
-
+      if (mVerbose) {
+        const auto dh = DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+        LOGP(info, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : ",
+             dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      }
       const auto* headerIn = DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
       auto payloadIn = ref.payload;
       auto payloadInSize = headerIn->payloadSize;
-      if (!mCompressedData) { //we have raw data coming in from flp
-        if (mVerbose) {
-          LOG(info) << " parsing non compressed data in the data reader task";
-        }
-
-        int a = 1;
-        int d = 1;
-        //        while(d==1){
-        //          a=sin(rand());
-        //        }
-
-        mReader.setDataBuffer(payloadIn);
-        mReader.setDataBufferSize(payloadInSize);
-        mReader.configure(mByteSwap, mVerbose, mHeaderVerbose, mDataVerbose);
-        if (mVerbose) {
-          LOG(info) << "%%% about to run " << loopcounter << " %%%";
-        }
-        mReader.run();
-        if (mVerbose) {
-          LOG(info) << "%%% finished running " << loopcounter << " %%%";
-        }
-        loopcounter++;
-        // mTracklets.insert(std::end(mTracklets), std::begin(mReader.getTracklets()), std::end(mReader.getTracklets()));
-        // mCompressedDigits.insert(std::end(mCompressedDigits), std::begin(mReader.getCompressedDigits()), std::end(mReader.getCompressedDigits()));
-        //mReader.clearall();
-        if (mVerbose) {
-          LOG(info) << "from parsing received: " << mTracklets.size() << " tracklets and " << mDigits.size() << " compressed digits";
-          LOG(info) << "relevant vectors to read : " << mReader.sumTrackletsFound() << " tracklets and " << mReader.sumDigitsFound() << " compressed digits";
+      if (std::string(headerIn->dataDescription.str) != std::string("DISTSUBTIMEFRAMEFLP")) {
+        if (!mCompressedData) { //we have raw data coming in from flp
```

Comment:
Now, that you have hardcoded RAWADAT, does this `mCompressedData` make sense? 
Which device is going to send compressed data?

Reply:
The idea was to make it full proof, one has to start the datareader task with a command line of option of  trd-datareader-compresseddata so the added safety of a different data type is not really required. You feel it should be done differently?

Reply:
Could you explain where the compressed data are coming from?

Reply:
datareader connects to crurawreader and compressedrawreader depending on the origin of its data.
So the CompressorTask uses the crurawreader and the DataReaderTask uses both depending data origin.
crurawreader-CompressorTask-send compressed to epn.

crurawreader/compressrawreader - DataReaderTask -- digits/tracklets/index--> ctf

my ascii art foo is weak.